Library/Formula/notmuch.rb

```diff
@@ -14,6 +14,7 @@ class Notmuch < Formula
 
   depends_on "pkg-config" => :build
   depends_on "emacs" => :optional
+  depends_on "python" => :optional
```

Comment:
This should be `depends_on :python => :optional`; the `:python` symbol has special powers; it uses the python in $PATH instead of always using Homebrew's python package (which is what the string form does).

Can you also add `depends_on :python3 => :optional`?